apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -151,21 +146,18 @@ resolve_loop(Begin, [], Fun, FailMsg, Zone) ->
 resolve_loop(Begin, Workers, Fun, FailMsg, Zone) ->
     receive
         {reply, Pid, Response} ->
-            {value, {Upstream, Pid, MonRef}, Workers0} =
-                lists:keytake(Pid, 2, Workers),
+            {value, {Pid, MonRef}, Workers0} =
+                lists:keytake(Pid, 1, Workers),
             _ = reply(Begin, Fun, Response, Zone),
             erlang:demonitor(MonRef, [flush]),
-            report_status([?MODULE, Upstream, successes]),
             resolve_loop(Begin, Workers0, skip, FailMsg, Zone);
         {'DOWN', MonRef, process, Pid, _Reason} ->
-            {value, {Upstream, Pid, MonRef}, Workers0} =
+            {value, {Pid, MonRef}, Workers0} =
                 lists:keytake(Pid, 2, Workers),
```

Comment:
```suggestion
                lists:keytake(Pid, 1, Workers),
```

Reply:
Oh. I did that fix, it crashes otherwise. :persevere:

Reply:
I will squash again on Github if you don't mind.